internal/pkg/composable/providers/kubernetessecrets/kubernetes_secrets.go

```diff
@@ -151,7 +151,8 @@ func (p *contextProviderK8SSecrets) Fetch(key string) (string, bool) {
 
 	if p.config.DisableCache {
 		// cache disabled - fetch secret from the API
-		return p.fetchFromAPI(ctx, secretName, secretNamespace, secretKey)
+		val, _, ok := p.fetchFromAPI(ctx, secretName, secretNamespace, secretKey)
```

Comment:
@pkoutsovasilis I'm guessing that adding logging here would add too much noise?

Reply:
Would a debug-level log make sense?

Reply:
https://github.com/elastic/elastic-agent/pull/7057 addressed in this PR 🙂